### PR TITLE
feat(console): unsaved changes alert

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(console): unsaved changes alert

### Note
React-Router V6 has removed the `useBlocker` & `usePrompt` API, so I implemented a workaround solution to support this feature. 
FYI: https://github.com/remix-run/react-router/issues/8139

### Related PRs
- [Refactor & Add unsaved changes alert for user settings](https://github.com/logto-io/logto/pull/1411)
- [Refactor & Add unsaved changes alert for connector details](https://github.com/logto-io/logto/pull/1414)

## Testing

SaveAlert

### CODE REVIEW

**Review by gao-sun — CHANGES_REQUESTED** [review]
your pnpm lock file looks not right

**Review by wangsijie — COMMENTED** [review]
return JSX in hooks is not a common practice?

**Review by wangsijie — APPROVED** [review]

File: packages/console/src/components/UnsavedChangesAlertModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import { Blocker, Transition } from 'history';
```
Comment (gao-sun) [inline packages/console/src/components/UnsavedChangesAlertModal/index.tsx:1]:
```suggestion
import type { Blocker, Transition } from 'history';
```

File: packages/phrases/src/locales/zh-cn.ts [hunk 2]
```diff
@@ -41,6 +41,9 @@ const translation = {
       add_another: '+ 新增',
       deletion_confirmation: '你确定要删除这个 {{title}} 吗?',
       settings_nav: '设置',
+      unsaved_changes_warning: '你有未保存的更改, 确定要离开吗？',
```
Comment (gao-sun) [inline packages/phrases/src/locales/zh-cn.ts:44]:
```suggestion
      unsaved_changes_warning: '还有未保存的变更, 确定要离开吗？',
```

**Review by gao-sun — APPROVED** [review]

**Pull request merged**